### ISSUE TITLE
NoClassDefFound Errors in Titan 09

### ISSUE THREAD

**Issue by superbeeny** [thinkaurelius/titan#828]
Hi all,

I pulled a copy of titan 09 down and tried to build it so I could access the maven artifacts. mvn clean install works nicely and all my dependencies resolve from my local maven repo.

When I try to run the code I get the following exception

```
Hello Titan!
0    [main] INFO  com.thinkaurelius.titan.diskstorage.cassandra.thrift.CassandraThriftStoreManager  - Closed Thrift connection pooler.
Exception in thread "main" java.lang.NoClassDefFoundError: org/objenesis/strategy/InstantiatorStrategy
    at com.thinkaurelius.titan.graphdb.database.serialize.kryo.KryoSerializer$2.initialValue(KryoSerializer.java:64)
    at com.thinkaurelius.titan.graphdb.database.serialize.kryo.KryoSerializer$2.initialValue(KryoSerializer.java:62)
    at java.lang.ThreadLocal.setInitialValue(ThreadLocal.java:180)
    at java.lang.ThreadLocal.get(ThreadLocal.java:170)
    at com.thinkaurelius.titan.graphdb.database.serialize.kryo.KryoSerializer.getKryo(KryoSerializer.java:77)
    at com.thinkaurelius.titan.graphdb.database.serialize.kryo.KryoSerializer.writeClassAndObject(KryoSerializer.java:132)
    at com.thinkaurelius.titan.graphdb.database.serialize.StandardSerializer$StandardDataOutput.writeClassAndObject(StandardSerializer.java:167)
    at com.thinkaurelius.titan.diskstorage.configuration.backend.KCVSConfiguration.object2StaticBuffer(KCVSConfiguration.java:246)
    at com.thinkaurelius.titan.diskstorage.configuration.backend.KCVSConfiguration.set(KCVSConfiguration.java:130)
    at com.thinkaurelius.titan.diskstorage.configuration.backend.KCVSConfiguration.set(KCVSConfiguration.java:120)
    at com.thinkaurelius.titan.diskstorage.configuration.ModifiableConfiguration.set(ModifiableConfiguration.java:27)
    at com.thinkaurelius.titan.graphdb.configuration.GraphDatabaseConfiguration.<init>(GraphDatabaseConfiguration.java:1280)
    at com.thinkaurelius.titan.core.TitanFactory.open(TitanFactory.java:91)
    at com.thinkaurelius.titan.core.TitanFactory.open(TitanFactory.java:71)
    at com.comcast.idea.titan.App.main(App.java:27)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:483)
    at com.intellij.rt.execution.application.AppMain.main(AppMain.java:134)
Caused by: java.lang.ClassNotFoundException: org.objenesis.strategy.InstantiatorStrategy
    at java.net.URLClassLoader$1.run(URLClassLoader.java:372)
    at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.net.URLClassLoader.findClass(URLClassLoader.java:360)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
    at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
    ... 20 more
```

Code

``` java
import com.thinkaurelius.titan.core.TitanFactory;
import com.thinkaurelius.titan.core.TitanGraph;
import org.apache.commons.configuration.BaseConfiguration;
import org.apache.commons.configuration.Configuration;

/**
 * Hello world!
 *
 */
public class App 
{
    public static Configuration getNicks09() {
        Configuration configuration = new BaseConfiguration();

        configuration.setProperty("storage.backend", "cassandrathrift");
        configuration.setProperty("storage.hostname", "localhost");

        return configuration;
    }

    public static void main( String[] args )
    {
        System.out.println( "Hello Titan!" );

        TitanGraph graph = TitanFactory.open(getNicks09());

        graph.close();
        System.out.println("Goodbye Titan!");
    }
}
```

Dependencies in pom

``` xml
<properties>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    <tinkerpopVersion>3.0.0.M5</tinkerpopVersion>
    <titanVersion>0.9.0-SNAPSHOT</titanVersion>
  </properties>

  <dependencies>
    <dependency>
      <groupId>com.thinkaurelius.titan</groupId>
      <artifactId>titan-core</artifactId>
      <version>${titanVersion}</version>
    </dependency>
    <dependency>
      <groupId>com.thinkaurelius.titan</groupId>
      <artifactId>titan-cassandra</artifactId>
      <version>${titanVersion}</version>
    </dependency>
    <dependency>
      <groupId>com.tinkerpop</groupId>
      <artifactId>gremlin-core</artifactId>
      <version>${tinkerpopVersion}</version>
    </dependency>
```

Dependency Tree from maven

```
[INFO] +- com.thinkaurelius.titan:titan-core:jar:0.9.0-SNAPSHOT:compile
[INFO] |  +- com.tinkerpop:gremlin-groovy:jar:3.0.0.M5:compile
[INFO] |  |  +- com.tinkerpop:gremlin-algorithm:jar:3.0.0.M5:compile
[INFO] |  |  +- org.apache.ivy:ivy:jar:2.3.0:compile
[INFO] |  |  +- org.codehaus.groovy:groovy-all:jar:indy:2.3.7:compile
[INFO] |  |  +- jline:jline:jar:2.11:compile
[INFO] |  |  +- org.apache.commons:commons-lang3:jar:3.3.1:compile
[INFO] |  |  \- org.kohsuke:groovy-sandbox:jar:1.6:compile
[INFO] |  +- org.glassfish:javax.json:jar:1.0:compile
[INFO] |  +- com.codahale.metrics:metrics-core:jar:3.0.1:compile
[INFO] |  +- com.codahale.metrics:metrics-ganglia:jar:3.0.1:compile
[INFO] |  |  \- info.ganglia.gmetric4j:gmetric4j:jar:1.0.3:compile
[INFO] |  |     \- org.acplt:oncrpc:jar:1.0.7:compile
[INFO] |  +- com.codahale.metrics:metrics-graphite:jar:3.0.1:compile
[INFO] |  +- org.reflections:reflections:jar:0.9.9-RC1:compile
[INFO] |  |  +- org.javassist:javassist:jar:3.16.1-GA:compile
[INFO] |  |  \- dom4j:dom4j:jar:1.6.1:compile
[INFO] |  |     \- xml-apis:xml-apis:jar:1.0.b2:compile
[INFO] |  +- com.spatial4j:spatial4j:jar:0.4.1:compile
[INFO] |  +- commons-collections:commons-collections:jar:3.2.1:compile
[INFO] |  +- commons-configuration:commons-configuration:jar:1.10:compile
[INFO] |  |  +- commons-lang:commons-lang:jar:2.6:compile
[INFO] |  |  \- commons-logging:commons-logging:jar:1.1.1:compile
[INFO] |  +- commons-io:commons-io:jar:2.1:compile
[INFO] |  +- commons-codec:commons-codec:jar:1.7:compile
[INFO] |  +- com.google.guava:guava:jar:15.0:compile
[INFO] |  +- com.google.code.findbugs:jsr305:jar:1.3.9:compile
[INFO] |  +- com.carrotsearch:hppc:jar:0.6.0:compile
[INFO] |  +- com.github.stephenc.high-scale-lib:high-scale-lib:jar:1.1.4:compile
[INFO] |  +- com.esotericsoftware.kryo:kryo:jar:2.24.0:compile
[INFO] |  |  \- com.esotericsoftware.minlog:minlog:jar:1.2:compile
[INFO] |  \- org.slf4j:slf4j-api:jar:1.7.5:compile
[INFO] +- com.thinkaurelius.titan:titan-cassandra:jar:0.9.0-SNAPSHOT:compile
[INFO] |  +- com.carrotsearch:junit-benchmarks:jar:0.7.0:compile
[INFO] |  +- com.carrotsearch.randomizedtesting:randomizedtesting-runner:jar:2.0.8:compile
[INFO] |  +- org.apache.cassandra:cassandra-all:jar:2.0.8:compile
[INFO] |  |  +- net.jpountz.lz4:lz4:jar:1.2.0:compile
[INFO] |  |  +- com.ning:compress-lzf:jar:0.8.4:compile
[INFO] |  |  +- commons-cli:commons-cli:jar:1.1:compile
[INFO] |  |  +- com.googlecode.concurrentlinkedhashmap:concurrentlinkedhashmap-lru:jar:1.3:compile
[INFO] |  |  +- org.antlr:antlr:jar:3.2:compile
[INFO] |  |  |  \- org.antlr:antlr-runtime:jar:3.2:compile
[INFO] |  |  |     \- org.antlr:stringtemplate:jar:3.2:compile
[INFO] |  |  |        \- antlr:antlr:jar:2.7.7:compile
[INFO] |  |  +- org.codehaus.jackson:jackson-core-asl:jar:1.9.2:compile
[INFO] |  |  +- org.codehaus.jackson:jackson-mapper-asl:jar:1.9.2:compile
[INFO] |  |  +- com.googlecode.json-simple:json-simple:jar:1.1:compile
[INFO] |  |  +- edu.stanford.ppl:snaptree:jar:0.1:compile
[INFO] |  |  +- org.mindrot:jbcrypt:jar:0.3m:compile
[INFO] |  |  +- com.yammer.metrics:metrics-core:jar:2.2.0:compile
[INFO] |  |  +- com.addthis.metrics:reporter-config:jar:2.1.0:compile
[INFO] |  |  |  \- org.hibernate:hibernate-validator:jar:4.3.0.Final:compile
[INFO] |  |  |     +- javax.validation:validation-api:jar:1.0.0.GA:compile
[INFO] |  |  |     \- org.jboss.logging:jboss-logging:jar:3.1.0.CR2:compile
[INFO] |  |  +- com.thinkaurelius.thrift:thrift-server:jar:0.3.3:compile
[INFO] |  |  |  \- com.lmax:disruptor:jar:3.0.1:compile
[INFO] |  |  +- net.sf.supercsv:super-csv:jar:2.1.0:compile
[INFO] |  |  +- log4j:log4j:jar:1.2.16:compile
[INFO] |  |  +- org.apache.thrift:libthrift:jar:0.9.1:compile
[INFO] |  |  |  +- org.apache.httpcomponents:httpclient:jar:4.2.5:compile
[INFO] |  |  |  \- org.apache.httpcomponents:httpcore:jar:4.2.4:compile
[INFO] |  |  +- org.apache.cassandra:cassandra-thrift:jar:2.0.8:compile
[INFO] |  |  +- com.github.stephenc:jamm:jar:0.2.5:compile
[INFO] |  |  \- io.netty:netty:jar:3.6.6.Final:compile
[INFO] |  +- commons-pool:commons-pool:jar:1.6:compile
[INFO] |  +- org.jboss.netty:netty:jar:3.2.7.Final:compile
[INFO] |  +- com.netflix.astyanax:astyanax-core:jar:1.56.48:compile
[INFO] |  |  +- com.eaio.uuid:uuid:jar:3.2:compile
[INFO] |  |  +- joda-time:joda-time:jar:1.6.2:compile
[INFO] |  |  \- org.codehaus.jettison:jettison:jar:1.2:compile
[INFO] |  |     \- stax:stax-api:jar:1.0.1:compile
[INFO] |  +- com.netflix.astyanax:astyanax-thrift:jar:1.56.48:compile
[INFO] |  +- com.netflix.astyanax:astyanax-cassandra:jar:1.56.48:compile
[INFO] |  |  \- org.apache.servicemix.bundles:org.apache.servicemix.bundles.commons-csv:jar:1.0-r706900_3:compile
[INFO] |  +- com.netflix.astyanax:astyanax-recipes:jar:1.56.48:compile
[INFO] |  \- org.xerial.snappy:snappy-java:jar:1.0.5-M3:compile
[INFO] +- com.tinkerpop:gremlin-core:jar:3.0.0.M5:compile
[INFO] |  +- org.yaml:snakeyaml:jar:1.13:compile
[INFO] |  +- org.javatuples:javatuples:jar:1.2:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.3.0:compile
[INFO] |  |  +- com.fasterxml.jackson.core:jackson-annotations:jar:2.3.0:compile
[INFO] |  |  \- com.fasterxml.jackson.core:jackson-core:jar:2.3.0:compile
[INFO] |  +- com.jcabi:jcabi-manifests:jar:1.0:compile
[INFO] |  |  \- com.jcabi:jcabi-log:jar:0.14:compile
[INFO] |  +- org.slf4j:slf4j-log4j12:jar:1.7.7:compile
[INFO] |  \- org.slf4j:jcl-over-slf4j:jar:1.7.7:compile
[INFO] \- junit:junit:jar:3.8.1:test
```

**Comment by superbeeny** [comment]
Appears adding 

http://mvnrepository.com/artifact/com.esotericsoftware.kryo/kryo/2.24.0

``` xml
    <dependency>
      <groupId>org.objenesis</groupId>
      <artifactId>objenesis</artifactId>
      <version>2.1</version>
    </dependency>
```

to my pom now resolves the issue.
Output now as expected:

```
Hello Titan!
0    [main] INFO  com.thinkaurelius.titan.graphdb.configuration.GraphDatabaseConfiguration  - Set cluster.partition=false from store features
59   [main] INFO  com.thinkaurelius.titan.graphdb.configuration.GraphDatabaseConfiguration  - Set default timestamp provider Timestamps[μs]
64   [main] INFO  com.thinkaurelius.titan.diskstorage.cassandra.thrift.CassandraThriftStoreManager  - Closed Thrift connection pooler.
99   [main] INFO  com.thinkaurelius.titan.graphdb.configuration.GraphDatabaseConfiguration  - Generated unique-instance-id=c0a801841128-HQSML-3116711
159  [main] INFO  com.thinkaurelius.titan.diskstorage.Backend  - Initiated backend operations thread pool of size 16
1385 [main] INFO  com.thinkaurelius.titan.diskstorage.log.kcvs.KCVSLog  - Loaded unidentified ReadMarker start time Timepoint[1415930156202000 μs] into com.thinkaurelius.titan.diskstorage.log.kcvs.KCVSLog$MessagePuller@23c30a20
1487 [main] INFO  com.thinkaurelius.titan.diskstorage.cassandra.thrift.CassandraThriftStoreManager  - Closed Thrift connection pooler.
Goodbye Titan!
```

**Comment by dalaro** [comment]
@superbeeny thanks for the very clean bug report.  There's a separate but related ASM issue in #821 (though they are distinct, not dupes).  Just wanted to link these in the tracker.

**Comment by dalaro** [comment]
This was fixed just before 0.9.0-M1 via 44632ff681fa2e95c2c0eaef9dc4d54ccb50196f.
